doc/source/debug.rst

```diff
@@ -0,0 +1,141 @@
+Debugging Kubernetes
+====================
+
+Sometimes your Kubernetes deployment doesn't behave the way you'd expect.
+This section provides some tips on debugging and fixing some common problems.
+
+Debugging commands
+------------------
+In order to debug your Kubernetes deployment, you need to be able to inspect
+the state of the pods are being used. The following are a few common commands
+for debugging.
+
+.. note::
+
+   You may need to add ``--namespace=<YOUR_NAMESPACE>`` in order for the
+   following commands to work.
+
+Let's say you've got a JupyterHub deployed, and a user tells you that they are
+experiencing strange behavior. Let's take a look at our deployment to figure
+out what is going on.
+
+``kubectl get pod``
+^^^^^^^^^^^^^^^^^^^
+In order to list all pods in your Kubernetes deployment, use the following
+command::
+
+    kubectl get pod
+
+This will output a list of all of the pods being used in the deployment. This
+includes two pods for the JupyterHub infrastructure (``hub-deployment`` and
+``proxy-deployment``) as well as any user pods that have been created as users
+log in. Here's an example of the output::
+
+    choldgraf@nhw2017-179318:~$ kubectl get pod --namespace=jhub
+    NAME                                READY     STATUS         RESTARTS   AGE
+    hub-deployment-3311438805-xnfvp     1/1       Running        0          2m
+    jupyter-choldgraf                   0/1       ErrImagePull   0          25s
+    proxy-deployment-1227971824-mn2wd   1/1       Running        0          5h
+
+Here we can see the two JupyterHub pods, as well as a single user pod. Note
+that all user pods will begin with ``jupyter-``.
+
+In particular, keep an eye on the ``STATUS`` column. If a given
+pod contains something other than ``Running``, then something may be wrong.
+
+In this case, we can see that our user's pod is in the ``ErrImagePull`` state.
+This generally means that there's something wrong with the Docker image that
+is defined in ``singleuser`` in our helm chart. Let's dig further...
+
+``kubectl describe pod``
+^^^^^^^^^^^^^^^^^^^^^^^^
+To see more detail about the state of a specific pod, use the following
+command::
+
+    kubectl describe pod <POD_NAME>
+
+This will output several pieces of information, including configuration and
+settings for the pod. The final section you'll see is a list of recent
+events. These can be particularly informative, as often an error will
+show up in this section. In our case, we notice that this is one of the

```

Comment:
New paragraph

**Real world scenario:** In our case, one of the lines in the events page displays an error